EIPS/eip-3436.md

```diff
@@ -0,0 +1,119 @@
+in-order chain exists and validator number 8 has just produced an in-turn block and then validators
+5, 7 and 8 go offline, leaving validators 1 to 6 to produce blocks. Two forks form, one with an
+in-order block from validator 1 and then an out of order block from validator 3. The second fork
+forms from validators 2, 4, and 6 in order. Both have a net total difficulty of 3 more than the
+common ancestor. So in this case if both forks become aware of the other fork then both are
+considered equally viable and neither set of validators should switch to the newly observed fork. In
+this case, adding a shortest chain rule would break the deadlock as the even numbered validators
+would adopt the shorter chain.
+
+For the second scenario with the same validator set and in-order chain with validator 7 having just
+produced an in order block, then validators 7 and 8 go offline. Two forks form, 1,3,5 on one side
+and 2,4,6 on the other. Both forks become aware of the other fork after producing their third block.
+In this case both forks have equal total difficulty and equal length. So Geth's rule would not break
+the tie and only the arrival of one of the missing validators fix the chain. In a worst case
+scenario the odd and even chains would produce a block for 7 and 8 respectively, and chain halt
+would result with no validators that have not chosen a fork. Only a manual rollback would fix this.
+
+One consideration when formulating the rules is that the block choice should be chosen so that it
+would encourage the maximum amount of in-order blocks. Selecting a chain based on shortest chain
+implicitly prefers the chain with more in-order blocks. When selecting between competing out of
+order chains the validator who is closest to producing an in-order block in the future should have
+their chain declined so that they are available to produce an in-order block sooner.
+
+At least one client has been observed producing multiple blocks at the same height with the same
+difficulty, so a final catch-all standard of lowest block hash should break any remaining ties.
+
+## Backwards Compatibility
+
+The current block choice rules are a mix of most total difficulty and most total difficulty plus
+shortest chain.
+
+As long as the majority of the active validators implement the block choice rules then a client who
+only implements the existing difficulty based rule will eventually align with the chain preferred by
+these rules. If less than a majority implement these rules then deadlocks can still occur, and
+depend on the first observation of problematic blocks, which is no worse than the current situation.
+
+If clients only partially implement the rule as long as every higher ranked rule is also implemented
+then the situation will be no worse than today.
+
+## Reference Implementation
+
+Implementation in Hyperledger Besu: https://github.com/hyperledger/besu/pull/2084
```

Comment:
Reference implementation should either be inline, in `../assets/eip-3436` or the entire section can be left out (it is optional).  EIPs are not an appropriate place for lists of things or external links to third party repositories.